bedrock/firefox/redirects.py

```diff
@@ -17,6 +17,37 @@ def firefox_channel(*args, **kwargs):
     return platform_redirector("firefox.channel.desktop", "firefox.channel.android", "firefox.channel.ios")
 
 
+def mobile_app(request, *args, **kwargs):
+    qs = request.META.get("QUERY_STRING", "")
+    campaign = None
+    product = "firefox"
+
+    product_options = ["firefox", "focus", "klar"]
+
+    campaign_options = [
+        "firefox-whatsnew",
+        "firefox-whatsnew-120",
+        "firefox-whatsnew-122",
+        "firefox-welcome-4",
+        "firefox-welcome-6",
+        "firefox-welcome-17",
+        "firefox-browsers-mobile-get-app",
+        "firefox-browsers-mobile-android",
+        "firefox-browsers-mobile-ios",
+        "firefox-browsers-mobile-focus",
+    ]
+
+    for p in product_options:
+        if f"product={p}" in qs:
+            product = p
+
+    for c in campaign_options:
+        if f"campaign={c}" in qs:
+            campaign = c
+
```

Comment:
This def works, but I think you could replace it with:

```
querydict = request.GET
product = querydict.get("product", "firefox")
campaign = querydict.get("campaign")
return mobile_app_redirector(request, product, campaign)
```

Reply:
Actually @alexgibson my suggestion doesn't work cos a querydict returns values in a list, even if theres just one key of a particular name, and dealing with that list (which may be empty) makes things a lot more complex. I say just leave it as you wrote it 👍